src/datastore/postgres-store.ts

```diff
@@ -2665,6 +2711,176 @@ export class PgDataStore extends (EventEmitter as { new (): DataStoreEventEmitte
     return { results };
   }
 
+  async updateNames(bnsName: DbBNSName) {
+    const {
+      name,
+      address,
+      registered_at,
+      expire_block,
+      zonefile_hash,
+      zonefile,
+      namespace_id,
+      latest,
+      tx_id,
+      status,
+      canonical,
+      index_block_hash,
+    } = bnsName;
+    const client = await this.pool.connect();
+    try {
+      await client.query('BEGIN');
+
+      await client.query(`UPDATE names SET latest = $1 WHERE name= $2`, [false, name]);
+
+      await client.query(
+        `
+        INSERT INTO names(
+          name, address, registered_at, expire_block, zonefile_hash, zonefile, namespace_id, latest, tx_id, status, canonical, index_block_hash
+        ) values($1, $2, $3, $4, $5, $6, $7, $8,$9, $10, $11, $12)
+        `,
+        [
+          name,
+          address,
+          registered_at,
+          expire_block,
+          zonefile_hash,
+          zonefile,
+          namespace_id,
+          latest,
+          tx_id,
+          status,
+          canonical,
+          index_block_hash,
+        ]
+      );
+
+      await client.query('COMMIT');
+    } catch (error) {
+      logError(`Error performing PG update: ${error}`, error);
+      await client.query('ROLLBACK');
+      throw error;
+    } finally {
+      client.release();
+    }
+  }
+
+  async updateNamespaces(bnsNamespace: DbBNSNamespace) {
```

Comment:
Same with this function -- should be called with an existing client connection

Reply:
Updated this as suggested